cache/cache.go

```diff
@@ -54,7 +48,7 @@ func (c Cache) Restore(src string) error {
 	return nil
 }
 
-func restoreCache(src string, s storage.Storage, a archive.Archive) error {
+func restoreCache(src string, fallback string, s storage.Storage, a archive.Archive) error {
```

Comment:
I feel like `restoreCache` should be dumb and just try and grab the file and restore it and the fallback should happen in `Restore`.